### PR TITLE
Pending requests interlibrary loan

### LINKED ISSUE #82: Pending Requests - Interlibrary Loan: Show pending items

### PR DESCRIPTION
**Pull request by erinesullivan**
# Overview
This pull request makes `Pending Requests > Interlibrary Loan` display items based on if the user has items where the `TransactionStatus` contains any of these words (according to [ILLiad Default Statuses & Queues](https://support.atlas-sys.com/hc/en-us/articles/360011812114-ILLiad-Default-Statuses-Queues) and personal testing):
- "Awaiting"
- "In Transit"
- "Pending"
- "Queue"
- "Request Sent"
- "Searching"
- "Submitted"

Closes #82.

## Cancel Link
A `Cancel Request` link has been added, which links to `https://ill.lib.umich.edu/illiad/illiad.dll?Action=21&Type=10&Value=<%=interlibrary_loan_request.illiad_id%>`.

## Title Hierarchy
The order of the title hierarchy has changed. When testing on my personal account (`esulliv`), it was showing the book chapter number as the title, and not the book title itself.

## Unit Tests
Unit tests have been added/broken up to help match the added classes.

## Testing
- Run the tests to make sure they pass: `docker-compose run web bundle exec rspec`
- Break the tests to double-check that they work
- In `./models/items/interlibrary_loan/interlibrary_loan_requests.rb` on `line 4` in the `actions` array, add the word `"Finished"`. Save and `docker-compose restart`. You should see about four items display.

### CODE REVIEW

File: spec/models/items/interlibrary_loan/interlibrary_loan_requests_spec.rb [hunk 1]
```diff
@@ -20,42 +20,8 @@
         subject.each do |item|
           items = items + item.class.name
         end
-        expect(items).to eq('InterlibraryLoanRequest')
+        expect(items).to eq('')
       end
     end
   end
 end
```
Comment (niquerio) [inline spec/models/items/interlibrary_loan/interlibrary_loan_requests_spec.rb:41]: I'd make sure these tests still work for one loan or one request. I'd load illiad_request.json into a variable in before(:each), explicitly modify the transaction type to match. Then load it in the stub.

File: views/interlibrary_loan_requests.erb [hunk 2]
```diff
@@ -10,14 +10,14 @@
 <table>
   <caption>
     <div class="loan-caption-inner-layout">
-      <p>Showing <strong><%= interlibrary_loan_requests.count %></strong> items</p>
+      <p>Showing <strong><%= interlibrary_loan_requests.count %></strong> <%= interlibrary_loan_requests.count == 1 ? "item" : "items" %></p>
```
Comment (niquerio) [inline views/interlibrary_loan_requests.erb]: This should a method in `models/items/items.rb`. maybe:
```
def item_text
  count ==1 ? "item" : "items"
end
```
And then write a corresponding test. (Which is easier to do in the model than for the view)

File: views/interlibrary_loan_requests.erb [hunk 3]
```diff
@@ -31,7 +31,7 @@
           <%= interlibrary_loan_request.creation_date %>
         </td>
         <td>
-          <%= interlibrary_loan_request.expiration_date %>
+          <a href="https://ill.lib.umich.edu/illiad/illiad.dll?Action=21&Type=10&Value=<%=interlibrary_loan_request.illiad_id%>">Cancel Request</a>
```
Comment (niquerio) [inline views/interlibrary_loan_requests.erb]: You could make the url a method in the request model, and a corresponding one in the loan model.

File: views/interlibrary_loans.erb [hunk 4]
```diff
@@ -32,7 +32,7 @@
         </td>
         <td>
           <% if interlibrary_loan.renewable? %>
-            <a class="button--light" href="https://ill.lib.umich.edu/illiad/illiad.dll?Action=11&Form=71&Value=<%=interlibrary_loan.illiad_id%>" disabled>Renew</a>
+            <a href="https://ill.lib.umich.edu/illiad/illiad.dll?Action=11&Form=71&Value=<%=interlibrary_loan.illiad_id%>">Request Renewal</a>
```
Comment (niquerio) [inline views/interlibrary_loans.erb]: See above. Making a url method in the loan model would make sense here.

**Review by niquerio — CHANGES_REQUESTED** [review]
Looking good. Just some small changes.

File: spec/models/items/interlibrary_loan/interlibrary_loan_requests_spec.rb [hunk 5]
```diff
@@ -2,16 +2,20 @@
 require 'json'
 
 describe InterlibraryLoanRequests do
-  context "one loan" do
+  context "one request" do
```
Comment (erinesullivan) [inline spec/models/items/interlibrary_loan/interlibrary_loan_requests_spec.rb]:
```suggestion
  context "two requests" do
```

**Review by niquerio — APPROVED** [review]

**Pull request merged**